configs/components/puppet.rb

```diff
@@ -96,9 +102,33 @@
     pkg.install_file("../osx_paths.txt", "/etc/paths.d/puppet-agent")
   end
 
+  if platform.is_windows?
+    pkg.environment "FACTERDIR" => settings[:prefix]
+  end
+
+  if platform.is_windows?
+    vardir = File.join(settings[:sysconfdir], 'puppet', 'cache')
+    configdir = File.join(settings[:sysconfdir], 'puppet', 'etc')
+  else
+    vardir = File.join(settings[:prefix], 'cache')
+    configdir = settings[:puppet_configdir]
+  end
   pkg.install do
     [
-      "#{settings[:host_ruby]} install.rb --ruby=#{File.join(settings[:bindir], 'ruby')} --no-check-prereqs --bindir=#{settings[:bindir]} --configdir=#{settings[:puppet_configdir]} --sitelibdir=#{settings[:ruby_vendordir]} --codedir=#{settings[:puppet_codedir]} --configs --quick --man --mandir=#{settings[:mandir]}",
+      "#{settings[:host_ruby]} install.rb \
+        --ruby=#{File.join(settings[:bindir], 'ruby')} \
+        --check-prereqs \
```

Comment:
Oh, this changed from no-check-prereqs to check-prereqs. That probably doesn't work for cross-compiles.